crates/libcontainer/src/seccomp/mod.rs

```diff
@@ -1,269 +1,65 @@
-                nix::errno::from_i32(res)
-            )
-        }
-        Ok(())
-    }
-}
-
-fn translate_syscall(syscall_name: &str) -> Result<i32> {
-    let c_syscall_name = CString::new(syscall_name)
-        .with_context(|| format!("Failed to convert syscall {:?} to cstring", syscall_name))?;
-    let res = unsafe { seccomp_syscall_resolve_name(c_syscall_name.as_ptr()) };
-    if res == __NR_SCMP_ERROR {
-        bail!("Failed to resolve syscall from name: {:?}", syscall_name);
+fn translate_arch(arch: Arch) -> ScmpArch {
+    match arch {
+        Arch::ScmpArchNative => ScmpArch::Native,
+        Arch::ScmpArchX86 => ScmpArch::X86,
+        Arch::ScmpArchX86_64 => ScmpArch::X8664,
+        Arch::ScmpArchX32 => ScmpArch::X32,
+        Arch::ScmpArchArm => ScmpArch::Arm,
+        Arch::ScmpArchAarch64 => ScmpArch::Aarch64,
+        Arch::ScmpArchMips => ScmpArch::Mips,
+        Arch::ScmpArchMips64 => ScmpArch::Mips64,
+        Arch::ScmpArchMips64n32 => ScmpArch::Mips64N32,
+        Arch::ScmpArchMipsel => ScmpArch::Mipsel,
+        Arch::ScmpArchMipsel64 => ScmpArch::Mipsel64,
+        Arch::ScmpArchMipsel64n32 => ScmpArch::Mipsel64N32,
+        Arch::ScmpArchPpc => ScmpArch::Ppc,
+        Arch::ScmpArchPpc64 => ScmpArch::Ppc64,
+        Arch::ScmpArchPpc64le => ScmpArch::Ppc64Le,
+        Arch::ScmpArchS390 => ScmpArch::S390,
+        Arch::ScmpArchS390x => ScmpArch::S390X,
     }
-
-    Ok(res)
 }
 
-fn translate_action(action: LinuxSeccompAction, errno: Option<u32>) -> u32 {
-    let errno = errno.unwrap_or(libc::EPERM as u32);
-    match action {
-        LinuxSeccompAction::ScmpActKill => SCMP_ACT_KILL,
-        LinuxSeccompAction::ScmpActTrap => SCMP_ACT_TRAP,
-        LinuxSeccompAction::ScmpActErrno => SCMP_ACT_ERRNO(errno),
-        LinuxSeccompAction::ScmpActTrace => SCMP_ACT_TRACE(errno),
-        LinuxSeccompAction::ScmpActAllow => SCMP_ACT_ALLOW,
-        LinuxSeccompAction::ScmpActKillProcess => SCMP_ACT_KILL_PROCESS,
-        LinuxSeccompAction::ScmpActNotify => SCMP_ACT_NOTIFY,
-        LinuxSeccompAction::ScmpActLog => SCMP_ACT_LOG,
-    }
+fn translate_action(action: LinuxSeccompAction, errno: Option<u32>) -> Result<ScmpAction> {
+    let errno = Some(errno.map(|e| e as i32).unwrap_or(libc::EPERM as i32));
+    let action_str = match action {
+        LinuxSeccompAction::ScmpActKill => "SCMP_ACT_KILL",
+        LinuxSeccompAction::ScmpActTrap => "SCMP_ACT_TRAP",
+        LinuxSeccompAction::ScmpActErrno => "SCMP_ACT_ERRNO",
+        LinuxSeccompAction::ScmpActTrace => "SCMP_ACT_TRACE",
+        LinuxSeccompAction::ScmpActAllow => "SCMP_ACT_ALLOW",
+        LinuxSeccompAction::ScmpActKillProcess => "SCMP_ACT_KILL_PROCESS",
+        LinuxSeccompAction::ScmpActNotify => "SCMP_ACT_NOTIFY",
+        LinuxSeccompAction::ScmpActLog => "SCMP_ACT_LOG",
+    };
+    ScmpAction::from_str(action_str, errno)
+        .with_context(|| format!("faild to tranlate ScmpAction from {action_str}"))
 }
```

Comment:
This could be simplified by modifying as follows.

```rust
fn translate_action(action: LinuxSeccompAction, errno: Option<u32>) -> Result<ScmpAction> {
    let errno = errno.map(|e| e as i32).unwrap_or(libc::EPERM as i32);                     
    let action = match action {                                                            
        LinuxSeccompAction::ScmpActKill => ScmpAction::KillThread,                         
        LinuxSeccompAction::ScmpActTrap => ScmpAction::Trap,                               
        LinuxSeccompAction::ScmpActErrno => ScmpAction::Errno(errno),                      
        LinuxSeccompAction::ScmpActTrace => ScmpAction::Trace(errno.try_into()?),          
        LinuxSeccompAction::ScmpActAllow => ScmpAction::Allow,                             
        LinuxSeccompAction::ScmpActKillProcess => ScmpAction::KillProcess,                 
        LinuxSeccompAction::ScmpActNotify => ScmpAction::Notify,                           
        LinuxSeccompAction::ScmpActLog => ScmpAction::Log,                                 
    };                                                                                     
                                                                                           
    Ok(action)                                                                             
}                                                                                                 
```

Reply:
@ManaSugi Thanks for your suggestion! I love your code. Can I ask you to use the suggestion feature of github because it can be able to include a commit in your name if you use it.

Reply:
Certainly. I fixed it using the feature. Thank you.

Reply:
Thanks 🙏